### ISSUE TITLE
Remove `iterators` from `MethodState`

### ISSUE THREAD

**Issue by B-rando1** [JacquesCarette/Drasil#3919]
As we found in a [discussion](https://github.com/JacquesCarette/Drasil/pull/3878#discussion_r1705735756) in #3878, the `iterators` field in `MethodState` will be unused once that PR is merged.  It is currently only used in one place in the C++ renderer, and @NoahCardoso changed the implementation so that it is no longer needed there.

Once #3878 is passed, we should do a quick double-check that `iterators` isn't used anywhere else; then we should be safe to remove it.

**Comment by NoahCardoso** [comment]
I fixed this issue. Once `Sets` gets merged I will make a pr for this.
